README.md

```diff
@@ -10,27 +10,52 @@ Create a file named `.github/workflows/cmd-action.yml` and add the following:
 ```yaml
 name: Command-Action
 
-on: 
+on:
   pull_request:
+  issue_comment:
+    types: [created]
 
 jobs:
   cmd-check:
     runs-on: ubuntu-latest
+    outputs:
+      commands: ${{ steps.cmd.outputs.commands }}
+      branch: ${{ steps.branch.outputs.branch }}
     name: Bot
     steps:
       - uses: actions/checkout@v4.1.1
+      # Switch to the PR branch if it's triggered by a comment
+      - run: gh pr checkout ${{ github.event.issue.number || github.event.pull_request.number }}
+        if: ${{ github.event.issue.pull_request }}
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
       - uses: paritytech/cmd-action@main
+        id: cmd
         with: 
           commands-directory: '.github/scripts'
-          GITHUB_TOKEN: ${{ github.token }}
+      - name: Export branch name
+        run: echo "branch=$(git rev-parse --abbrev-ref HEAD)" >> "$GITHUB_OUTPUT"
+        id: branch
+
+  cmd-run:
+    needs: [cmd-check]
+    if: ${{ github.event.issue.pull_request }}
+    continue-on-error: true
+    # We set the current machine here (to differ between ours and generic ones)
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        command: ${{ fromJson(needs.cmd-check.outputs.commands) }}
+    name: Run command
+    steps:
+      - uses: paritytech/cmd-action/run@main
+        with:
+          branch: ${{ needs.cmd-check.outputs.branch }}
+          command: ${{ matrix.command }}
```

Comment:
This sounds a bit extensive for the repository-side action. Sounds like something we will have to update frequently.  
Can we have this code inside of this repo, and just set the parameters?

Reply:
I‘m afraid I‘m not following you. Could you please rephrase it?

Reply:
I meant the whole action. There's logic in here, with checking out the PR branch, doing some calculations. I'd expect something more simple to be put in the repositories that use the action. 

We are already reusing `paritytech/cmd-action/run@main`, is it possible to move some of this logic in this repo, and not spread it outside?

Reply:
I now simplified it into it's own composite action.

Except the git push as that can be dependent on the docker image and I believe it's better to let the author customize it.